backend/app/src/main/java/com/ugent/pidgeon/controllers/ClusterController.java

```diff
@@ -168,6 +173,68 @@ public ResponseEntity<?> doGroupClusterUpdate(GroupClusterEntity clusterEntity,
         return ResponseEntity.ok(entityToJsonConverter.clusterEntityToClusterJson(clusterEntity));
     }
 
+    /**
+     * Fills up the groups in a cluster by providing a map of groupids with lists of userids
+     *
+     * @param clusterid  identifier of a cluster
+     * @param auth authentication object of the requesting user
+     * @param clusterFillJson ClusterFillJson object containing a map of all groups and their
+     *                        members of that cluster
+     * @return ResponseEntity<?>
+     * @HttpMethod PUT
+     * @ApiPath /api/clusters/{clusterid}/fill
+     * @AllowedRoles student, teacher
+     */
+    @PutMapping(ApiRoutes.CLUSTER_BASE_PATH + "/{clusterid}/fill")
+    @Roles({UserRole.teacher, UserRole.student})
+    public ResponseEntity<?> fillCluster(@PathVariable("clusterid") Long clusterid, Auth auth, @RequestBody ClusterFillJson clusterFillJson) {
+        CheckResult<GroupClusterEntity> checkResult = clusterUtil.getGroupClusterEntityIfAdminAndNotIndividual(clusterid, auth.getUserEntity());
+
+        if (checkResult.getStatus() != HttpStatus.OK) {
+            return ResponseEntity.status(checkResult.getStatus()).body(checkResult.getMessage());
+        }
+
+        GroupClusterEntity groupCluster = checkResult.getData();
+
+        ResponseEntity<?> response = getCluster(groupCluster.getId(), auth);
+        if(response.getStatusCode() != HttpStatus.OK){
+            return response;
+        }
+
+        GroupClusterJson clusterJson = (GroupClusterJson) response.getBody();
+        if(clusterJson == null){
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Group cluster could not be found");
+        }
+
+        if(clusterFillJson.getClusterGroupMembers().keySet().size() > clusterJson.groupCount()){
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("provided more groups than are allowed in the cluster");
+        }
+
+        if(clusterFillJson.getClusterGroupMembers().values().stream().anyMatch(members -> members.length > clusterJson.capacity())){
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("you made a group with too many members");
```

Comment:
Ik zou dus voor elke groep checken in de clusterFillJson checken of

1. de groep een valid groep is & gekoppeld aan die cluster
2. er niet meer members zijn dan de maxSize
3. alle users valid users zijn die in de course zitten

Eenmaal dit voor alle groepen id request gecheckt is kunnen ze toegevoegd worden zoals je dit reeds doet